pallets/asset-index/src/lib.rs

```diff
@@ -182,6 +196,56 @@ pub mod pallet {
             Ok(().into())
         }
 
+        #[pallet::weight(10_000)] // TODO: Set weights
+        /// Callable by an admin to remove assets from the index
+        /// and maybe transfer some IndexToken
+        ///
+        /// Caller balance and recipient are updated to allocate
+        /// the correct amount of the IndexToken
+        pub fn remove_asset(
+            origin: OriginFor<T>,
+            asset_id: T::AssetId,
+            units: T::Balance,
+            recipient: Option<MultiLocation>,
+            recipient_account: Option<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            T::AdminOrigin::ensure_origin(origin.clone())?;
+            let caller = ensure_signed(origin)?;
+            <Self as AssetRecorder<T::AssetId, T::Balance>>::remove_asset(
+                &asset_id, &units, recipient,
+            )?;
```

Comment:
This modifies while following operations are fallible, so we need to either mark this extrinsic make it mutation safe.

Reply:
The `remove_asset` call only contains this one mutating storage operation, and it runs at the top of the call `remove_asset`